packages/components/src/internal/components/samples/SamplesAssayButton.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { FC, memo } from 'react';
+import { DropdownButton } from 'react-bootstrap';
+import { PermissionTypes } from '@labkey/api';
+
+import { RequiresPermission } from '../../..';
+
+import { QueryModel } from '../../../public/QueryModel/QueryModel';
+
+import { AssayImportSubMenuItem } from '../assay/AssayImportSubMenuItem';
+
+import { isSamplesSchema } from './utils';
+
+interface Props {
+    model: QueryModel;
+    providerType?: string;
+    asSubMenu?: boolean;
+}
+
+export const SamplesAssayButton: FC<Props> = memo(props => {
+    const { model, providerType, asSubMenu } = props;
+
+    if (!isSamplesSchema(model?.schemaQuery)) return null;
```

Comment:
I wonder when this can happen.

Reply:
I think there might be one or two of the "All Samples" tab cases that hit this case and don't show the Assay menu because of it. I think it is an open issue somewhere or behavior we do want to change/allow, but not as part of this story.